File: sr_recognizer/src/recognizer_client.py
#! /usr/bin/env python

from __future__ import print_function
import rospy


# Brings in the SimpleActionClient
import actionlib

# Brings in the messages used by the fibonacci action, including the
# goal message and the result message.
import sr_recognizer.msg

def recognizer_client():
    # Creates the SimpleActionClient, passing the type of the action
    # (FibonacciAction) to the constructor.
    client = actionlib.SimpleActionClient('recognizer', recognizer.msg.RecognizerAction)

    # Waits until the action server has started up and started
    # listening for goals.
    client.wait_for_server()

    # Creates a goal to send to the action server.
    goal = recognizer.msg.RecognizerGoal(order=20)

    # Sends the goal to the action server.
    client.send_goal(goal)

    # Waits for the server to finish performing the action.
    client.wait_for_result()

    # Prints out the result of executing the action
    return client.get_result()  # A FibonacciResult

if __name__ == '__main__':
    try:
        # Initializes a rospy node so that the SimpleActionClient can
        # publish and subscribe over ROS.
        rospy.init_node('recognizer_client_py')
        result = recognizer_client()

        result_names = list()

        for i in xrange(0, len(result.ids)):
            result_names.append(result.ids[i].data)

        print("Result Names:", result_names)

    
        tt = result.transforms
        print("Transforms:\n", tt)
     
    except rospy.ROSInterruptException:
        print("program interrupted before completion", file=sys.stderr)

File: sr_recognizer/src/recognizer_client.cpp
#include <iostream> 
 
#include <ros/ros.h> 
#include <actionlib/client/simple_action_client.h> 
#include <actionlib/client/terminal_state.h> 
#include <sr_recognizer/RecognizerAction.h> 
 
 
int main (int argc, char **argv) 
{ 
  ros::init(argc, argv, "test_Recognizer"); 
 
  // create the action client 
  // true causes the client to spin its own thread 
  actionlib::SimpleActionClient<sr_recognizer::RecognizerAction> ac("recognizer", true); 
 
  ROS_INFO("Waiting for action server to start."); 
  // wait for the action server to start 
  ac.waitForServer(); //will wait for infinite time 
 
  ROS_INFO("Action server started, sending goal."); 
  // send a goal to the action 
  sr_recognizer::RecognizerGoal goal; 
  goal.start = 1; 
  ac.sendGoal(goal); 
 
  //wait for the action to return 
  bool finished_before_timeout = ac.waitForResult(ros::Duration(600.0)); 
 
 
  if (finished_before_timeout) 
  { 
    actionlib::SimpleClientGoalState state = ac.getState(); 
 
    sr_recognizer::RecognizerResult result_ = *ac.getResult(); 
 
    ROS_INFO("Recognized Objects: %s", result_.ids[0].data.c_str()); 
 
    ROS_INFO("Action finished: %s",state.toString().c_str()); 
  } 
  else 
    ROS_INFO("Action did not finish before the time out."); 
 
  //exit 
  return 0; 
} 
